app/models/cart.rb

```diff
@@ -0,0 +1,21 @@
+class Cart < ApplicationRecord
+  belongs_to :user
+  has_many :cart_products, dependent: :destroy
+  has_many :products, through: :cart_products
+
+  enum status: {
+    in_process: 0,
+    completed: 1
+  }
+
+  validates :total_items, :total_price, numericality: { greater_than_or_equal_to: 0 },
+                                        presence: true
+
+  def calculate_total_items
```

Comment:
Sounds great, but by including this gem for the functionality I'm implementing at the model level, it would break the scheme of the decorator pattern. At the moment these instance methods are required to update values in the columns of the `Cart` table.